datafusion/core/tests/fuzz_cases/join_fuzz.rs

```diff
@@ -217,6 +225,7 @@ async fn test_semi_join_1k() {
 
 #[tokio::test]
 async fn test_semi_join_1k_filtered() {
+    // NLJ vs HJ gives wrong result
```

Comment:
The HashJoin and NestedLoop Join gives different results for filtered joins fuzz tests. 
In the same time HashJoin and SortMergeJoin results are correct. 

Need to triage what is going on.

Reply:
Could you add a reference to the issue with the comment here